### PR TITLE
Add New Port ASSAULT SHELL

### PR DESCRIPTION
**Pull request by MadShmupper**
Add New Port ASSAULT SHELL

# New Port for { ASSAULT SHELL }

## Game Information
- **Title**: { ASSAULT SHELL } 
- **URL**: {https://store.steampowered.com/app/1432970/ASSAULT_SHELL/}

### CFW Tests
Ensure your game has been tested on all major CFWs:
- [X] AmberELEC
- [X] ArkOS
- [X] ROCKNIX
- [X] muOS

### Resolution Tests
Test all major resolutions:
- [X] 480x320
- [X] 640x480
- [X] 720x720 (RGB30)
- [X] Higher resolutions (e.g., 1280x720)

## File Structure
- Your port should have the following structure:
  - portname/
    - port.json
    - README.md
    - screenshot.jpg
    - cover.jpg
    - gameinfo.xml
    - Port Name.sh
    - portname/
      - <portfiles here>

## Additional Resources
For an in-depth guide on creating a pull request, refer to: [PortMaster Game Packaging Guide](https://portmaster.games/packaging.html#creating-a-pull-request)

### CODE REVIEW

File: ports/assaultshell/ASSAULT SHELL.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+elif [ -d "/opt/tools/PortMaster/" ]; then
+  controlfolder="/opt/tools/PortMaster"
+elif [ -d "$XDG_DATA_HOME/PortMaster/" ]; then
+  controlfolder="$XDG_DATA_HOME/PortMaster"
+else
+  controlfolder="/roms/ports/PortMaster"
+fi
+
+source $controlfolder/control.txt
+source $controlfolder/device_info.txt
```
Comment (kloptops) [inline ports/assaultshell/ASSAULT SHELL.sh:16]: This is no longer needed.

```suggestion
```

File: ports/assaultshell/ASSAULT SHELL.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+export PORT_32BIT="Y"
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
+get_controls
+
+$ESUDO chmod 666 /dev/tty0
```
Comment (kloptops) [inline ports/assaultshell/ASSAULT SHELL.sh:21]: Also this:
```suggestion
```

File: ports/assaultshell/ASSAULT SHELL.sh [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
+
+[ -f "./gamedata/data.win" ] && mv gamedata/data.win gamedata/game.droid
+
+$GPTOKEYB "gmloader" &
+echo "Loading, please wait... " > /dev/tty0
+
+$ESUDO chmod +x "$GAMEDIR/gmloader"
+
+./gmloader assaultshell.apk
```
Comment (kloptops) [inline ports/assaultshell/ASSAULT SHELL.sh:40]: Missing `pm_platform_helper` call.

```suggestion
pm_platform_helper "$GAMEDIR/assaultshell.apk"
./gmloader assaultshell.apk
```

File: ports/assaultshell/ASSAULT SHELL.sh [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+$ESUDO kill -9 "$(pidof gptokeyb)"
```
Comment (kloptops) [inline ports/assaultshell/ASSAULT SHELL.sh:42]: The rest of the script can be replaced with this:

```suggestion
pm_finish
```

File: ports/assaultshell/ASSAULT SHELL.sh [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+$ESUDO systemctl restart oga_events &
```
Comment (kloptops) [inline ports/assaultshell/ASSAULT SHELL.sh:43]:
```suggestion
```

File: ports/assaultshell/ASSAULT SHELL.sh [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+printf "\033c" >> /dev/tty1
```
Comment (kloptops) [inline ports/assaultshell/ASSAULT SHELL.sh:44]:
```suggestion
```

File: ports/assaultshell/ASSAULT SHELL.sh [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+printf "\033c" > /dev/tty0
```
Comment (kloptops) [inline ports/assaultshell/ASSAULT SHELL.sh:45]:
```suggestion
```

**Review by kloptops — DISMISSED** [review]

File: ports/assaultshell/ASSAULT SHELL.sh (hunk 3 above)
Comment (cdeletre) [reply ports/assaultshell/ASSAULT SHELL.sh:40]: `pm_platform_helper "$GAMEDIR/gmloader"` will be better :)

File: ports/assaultshell/screenshot.png
Comment (cdeletre) [inline ports/assaultshell/screenshot.png:1]: File size can be optimized using `pngquant` or an online tool.

File: ports/assaultshell/cover.png
Comment (cdeletre) [inline ports/assaultshell/cover.png:1]: File size can be optimized using `pngquant` or an online tool.

File: ports/assaultshell/port.json [hunk 8]
```diff
@@ -0,0 +1,31 @@
+{
+  "version": 3,
+  "name": "assaultshell.zip",
+  "items": [
+    "ASSAULT SHELL.sh",
+    "assaultshell/"
```
Comment (cdeletre) [inline ports/assaultshell/port.json:6]:
```suggestion
    "assaultshell"
```

**Review by cdeletre — CHANGES_REQUESTED** [review]

File: ports/assaultshell/port.json (hunk 8 above)
Comment (MadShmupper) [reply ports/assaultshell/port.json:6]: The / is literally in the port.json example.

**Comment by Cebion** [comment]
Still needs some changes

File: ports/assaultshell/cover.png
Comment (MadShmupper) [reply ports/assaultshell/cover.png:1]: I shrunk the png but it didn't shave much using IrfanView with a 9 comp-level🙂

File: ports/assaultshell/screenshot.png
Comment (MadShmupper) [reply ports/assaultshell/screenshot.png:1]: Ayyyy..🙂 switched screenshot.png to a .jpg and shaved a bunch.
Ty!🎩

File: ports/assaultshell/port.json (hunk 8 above)
Comment (MadShmupper) [reply ports/assaultshell/port.json:6]: I totally don't mind taking it out if we all agree it's out but the example must be changed🙂

File: ports/assaultshell/ASSAULT SHELL.sh (hunk 3 above)
Comment (MadShmupper) [reply ports/assaultshell/ASSAULT SHELL.sh:37]: This has been updated. I do apologize for not resolving sooner😅🎩ty for hanging in there😂

**Review by cdeletre — APPROVED** [review]

**Pull request merged**